src/engine/Server.h

```diff
@@ -150,4 +156,28 @@ class Server {
       const ad_utility::websocket::QueryId& queryId,
       const std::shared_ptr<Operation>& rootOperation,
       std::optional<std::chrono::seconds> timeLimit) const;
+
+  /// Parse the query and plan it on `threadPool_`.
+  net::awaitable<PlannedQuery> parseAndPlan(const std::string& query,
+                                            QueryExecutionContext& qec) const;
+
+  /// Check if the access token is valid. Return true if the access token
+  /// exists and is valid. Return false if there's no access token passed.
+  /// Throw an exception if there is a token passed but it doesn't match,
+  /// or there is no access token set by the server config.
```

Comment:
Also add that the error message of the thrown exception is formulated towards end users, so you can directly  send it as the text of a 4xx or 5xx repsonse.